internal/helm/chart/dependency_manager_test.go

```diff
@@ -29,12 +30,38 @@ import (
 	. "github.com/onsi/gomega"
 	helmchart "helm.sh/helm/v3/pkg/chart"
 	helmgetter "helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/registry"
 	"helm.sh/helm/v3/pkg/repo"
 
 	"github.com/fluxcd/source-controller/internal/helm/chart/secureloader"
 	"github.com/fluxcd/source-controller/internal/helm/repository"
 )
 
+type mockTagsGetter struct {
+	tags map[string][]string
+}
+
+func (m *mockTagsGetter) Tags(requestURL string) ([]string, error) {
+	u, err := url.Parse(requestURL)
+	if err != nil {
+		return nil, err
+	}
+
+	name := filepath.Base(u.Path)
+	if tags, ok := m.tags[name]; ok {
+		return tags, nil
+	}
+	return nil, fmt.Errorf("no tags found for %s", name)
```

Comment:
```suggestion
	return nil, fmt.Errorf("no tags found for %s", requestURL)
```
This is what's provided by callers so the error should use that instead of the internally generated path. Maybe you can add the path as an additional parameter of the format string for easier debugging.

Reply:
I see. So yeah, I would prefer to have the input of the func be part of the error so let's maybe just output both?!